extensions/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentPuller.java

```diff
@@ -202,8 +205,8 @@ public InputStream openStream() throws IOException
         return result;
       }
       if (CompressionUtils.isGz(s3Coords.path)) {
-        final String fname = Paths.get(uri).getFileName().toString();
-        final File outFile = new File(outDir, CompressionUtils.getGzBaseName(fname));
+        final String fname = Files.getNameWithoutExtension(uri.getPath());
```

Comment:
This is probably actually a "bug" in the current stable, but we don't have anything that pulls gzip off s3, so it is never encountered.